### PR TITLE
feat: add NavStack type

### LINKED ISSUE #61: Fix GoBack
Problem : 

Ideally GoBack should go back to previous context , but right now its not implemented properly and it goes to main-menu 

Solution : 
Implement an abstraction to jump to previous context when we want to go back

### PR DESCRIPTION
**Pull request by BretasArthur1**
## Summary

Adds navigation types to handle in-section navigation.

## What's in here

- `Section` enum with per-section max depth
- `SectionNav` for tracking depth within a section (bounded, no heap)
- `AppNav` for managing main menu vs in-section state
- Tests covering normal flow and edge cases

Next step: integrate into the command flow.

First step to close #61

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ pub mod error;
 pub mod misc;
 pub mod prompt;
 pub mod ui;
+pub mod types;
```
Comment (Nagaprasadvr) [inline src/main.rs]: can add this inside misc/navigation.rs

**Review by sonicfromnewyoke — APPROVED** [review]
LGTM, what do you think about using **type state pattern** here?
- https://cliffle.com/blog/rust-typestate
- https://zerotomastery.io/blog/rust-typestate-patterns

to me navigation is a perfect state machine. In the first link you can find references to [serde](https://docs.rs/serde/latest/serde/ser/trait.Serializer.html)

**Comment by BretasArthur1** [comment]
> LGTM, what do you think about using **type state pattern** here?
> 
> - https://cliffle.com/blog/rust-typestate
> 
> - https://zerotomastery.io/blog/rust-typestate-patterns
> 
> 
> 
> to me navigation is a perfect state machine. In the first link you can find references to [serde](https://docs.rs/serde/latest/serde/ser/trait.Serializer.html) 


Perfect! Agree with you, navigatio is a perfect state machine scenario. I tried to use this as my base as well.


And I'll take a look on this references and open to change the implementation based on a better pattern. This one was based on my own thought process actually. I'll check and see what I can abstract to use here, thx for the suggestion 🤝

File: src/types.rs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+    Config,
+    Stake,
+    Transaction,
+    Vote,
+}
+
+impl Section {
+    pub const fn max_depth(self) -> usize {
+        match self {
+            Section::Account => 6,
```
Comment (Nagaprasadvr) [inline src/types.rs]: ig the current depth is 1

File: src/misc/navigation.rs [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+    pub const fn max_depth(self) -> usize {
+        match self {
+            CommandSection::Cluster => 1,     // 0 Prompts
+            CommandSection::Account => 2,     // 1 Prompt
+            CommandSection::Config => 3,      // 2 Prompts
+            CommandSection::Transaction => 3, // 2 Prompts
+            CommandSection::Vote => 5,        // 4 Prompts
+            CommandSection::Stake => 8,       // 7 Prompts
+        }
+    }
```
[the remark this answers is not in the document]
Comment (BretasArthur1) [reply src/misc/navigation.rs]: Ok, let me know your thoughts

**Comment by BretasArthur1** [comment]
Closed because I'll change some of the implementation. Create it simplified
